Kernel/ACPI/Definitions.h

```diff
@@ -109,28 +116,44 @@ struct FADT {
     GenericAddressStructure sleep_control;
     GenericAddressStructure sleep_status;
     u64 hypervisor_vendor_identity;
+};
 
-} __attribute__((__packed__));
+struct [[gnu::packed]] MADTEntry
+{
+    u8 type;
+    u8 length;
+    char data[];
+};
 
-struct MADT : public SDTHeader {
+struct [[gnu::packed]] MADT
+{
+    SDTHeader h;
+    u32 lapic_address;
+    u32 flags;
+    MADTEntry entries[];
 };
 
-struct DSDT : public SDTHeader {
+struct [[gnu::packed]] AMLTable
+{
+    SDTHeader h;
+    char aml_code[];
 };
 
-struct PCI_MMIO_Descriptor {
+struct [[gnu::packed]] PCI_MMIO_Descriptor
```

Comment:
It's okay to change from `__attribute__((__packed__))` to `[[gnu::packed]]` but please do that sort of thing in a separate commit, since it makes the actual changes much harder to read.

Reply:
Sure, will do so :)